lib/rubocop/ast/processed_source.rb

```diff
@@ -227,61 +236,77 @@ def tokenize(parser)
       end
 
       # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength
-      def parser_class(ruby_version)
-        case ruby_version
-        when 1.9
-          require 'parser/ruby19'
-          Parser::Ruby19
-        when 2.0
-          require 'parser/ruby20'
-          Parser::Ruby20
-        when 2.1
-          require 'parser/ruby21'
-          Parser::Ruby21
-        when 2.2
-          require 'parser/ruby22'
-          Parser::Ruby22
-        when 2.3
-          require 'parser/ruby23'
-          Parser::Ruby23
-        when 2.4
-          require 'parser/ruby24'
-          Parser::Ruby24
-        when 2.5
-          require 'parser/ruby25'
-          Parser::Ruby25
-        when 2.6
-          require 'parser/ruby26'
-          Parser::Ruby26
-        when 2.7
-          require 'parser/ruby27'
-          Parser::Ruby27
-        when 2.8, 3.0
-          require 'parser/ruby30'
-          Parser::Ruby30
-        when 3.1
-          require 'parser/ruby31'
-          Parser::Ruby31
-        when 3.2
-          require 'parser/ruby32'
-          Parser::Ruby32
-        when 3.3
-          require 'parser/ruby33'
-          Parser::Ruby33
-        when 3.4
-          require 'parser/ruby34'
-          Parser::Ruby34
-        else
-          raise ArgumentError,
-                "RuboCop found unknown Ruby version: #{ruby_version.inspect}"
+      def parser_class(ruby_version, parser_engine)
+        case parser_engine
+        when :parser
```

Comment:
I wonder if we shouldn't name this `:parser_whitequark`, give that `parser` is a very generic name.

Reply:
`parser_whitequark` is a unique name, but the context might become distant if the org of `whitequark/parser` changes to something other than whitequark.

Maybe something like `parser_gem` can be suggested, but I'm not that confident about that. So, `parser_whitequark` is fine too, if @bbatsov prefers that :-)

Reply:
For now, I've renamed it to `parser_whitequark`. Please let me know if you'd like the name updated further.

Reply:
I think that name is fine and we can always provide some aliases down the road. 

In general, I'm guessing that at some point native Prism (without the `parser_whitequark`) translation will be the only parsing engine we'll be supporting. I guess that once most tools switch to Prism maintaining the older parsers will become pointless. It's amazing how quickly Prism is getting traction in the Ruby community these days https://blog.jruby.org/2024/02/23/jruby-prism-parser.html